pkg/dependencies/factory.go

```diff
@@ -596,39 +595,15 @@ func (f *Factory) WithTroubleshoot() *Factory {
 	return f
 }
 
-func (f *Factory) WithHelmSecure() *Factory {
+func (f *Factory) WithHelm(insecure bool) *Factory {
```

Comment:
can we make it optional arg? 
```suggestion
func (f *Factory) WithHelm(opts ...HelmOpts) *Factory {
```